design_documents/parametrizing_tests_with_fixed_values.md

```diff
@@ -0,0 +1,168 @@
+# Parametrizing Tests With Fixed Values
+
+## Context
+
+Various test runners for other programming languages provide ways of parametrizing test cases with fixed values.
+This way, multiple test cases can be executed for the same test source code.
+
+## Goal
+
+Propose a way of adding support for parametrized tests to Starknet Foundry's `snforge`.
+
+## Proposed Solution
+
+Introduce two new attributes: `parametrize` and `case`.
+
+- `parametrize` - defines a parametrized test.
+- `case` - defines a specific test case
+
+### Parametrization With Simple Arguments
+
+```cairo
+#[parametrize]
+#[case(1, 3)]
+#[case(3, 5)]
+#[test]
+fn my_test(a: felt252, b: u32) {
+    // case a = 1, b = 3
+    // case a = 3, b = 5
+    // ...
+}
+```
+
+For test cases like this, values could simply be injected into a runner for each `case`.
+
+### Parametrization With Fixtures
+
+```cairo
+struct MyStruct {
+    a: felt252, 
+    b: Array<felt252>
+}
+
+fn my_fixture() -> MyStruct {
+    MyStruct { a: 1, b: array![2, 3] }
+}
+
+#[parametrize]
+#[case(1, my_fixture())]
+#[case(3, my_fixture())]
```

Comment:
Would it be possible to create a struct directly in the `case`?

Reply:
Syntax like 

```cairo
#[case(MyStruct { a: 1, b: array![2, 3] })]`
```

should be valid in cairo. However from what I understand, attributes don't get type-checked, and we'd have to manually handle this struct creation.

I wonder now if this could be just handled with code-gen as well.